source/kraken/maintenance_worker.cpp

```diff
@@ -56,14 +56,130 @@ namespace bg = boost::gregorian;
 
 namespace navitia {
 
+void MaintenanceWorker::run() {
+    // try to load data from disk
+    load_data();
+
+    if (!is_data_loaded()) {
+        // The data is not loaded, so we wait for a reload message on the task queue
+        // that results in a successfull data load.
+        // We may be disconnected from rabbitmq during this wait,
+        // and reconnection to rabbitmq is handled by this while(true) loop
+        while (true) {
```

Comment:
is this loop still necessary since we use `libkeepalive` now?

Reply:
it seems wether the data is loaded or not, you need to open channel to rabbitmq, can you do that once at the beginning?

Reply:
- the loop is necessary to handle disconnection on rabbitmq side, like the rabbitmq service died for some reason
- when rabbitmq dies, we need to reopen the channel, that's why it is in the loop